test262/per-file/main.js

```diff
@@ -0,0 +1,225 @@
+"use strict";
+
+// (() => {
+let resultsNode;
+let summaryLabel;
+let summaryStatusLabel;
+let leafTreeNodeTemplate;
+let nonLeafTreeNodeTemplate;
+let pathInTree = [];
+let tree;
+const resultsObject = Object.create(null);
+
+function initialize(data, modeName) {
+  let mode;
+  if (modeName === "") {
+    mode = "AST";
+  } else if (modeName === "-bytecode") {
+    mode = "bytecode";
+  } else {
+    throw new Error("Unknown mode: " + modeName);
+  }
+
+  // Do a pass and generate the tree.
+  for (const testPath in data.results) {
+    const segments = testPath.split("/");
+    const fileName = segments.pop();
+    let testObject = resultsObject;
+    for (const pathSegment of segments) {
+      if (!(pathSegment in testObject)) {
+        testObject[pathSegment] = {
+          children: Object.create(null),
+          aggregatedResults: null,
+        };
+      }
+
+      testObject = testObject[pathSegment].children;
+    }
+
+    if (!(fileName in testObject)) {
+      testObject[fileName] = {
+        children: null,
+        results: Object.create(null),
+      };
+    }
+
+    testObject[fileName].results[mode] = data.results[testPath].toLowerCase();
+  }
+}
+
+function generateResults() {
+  function constructTree(results) {
+    if (results.children === null) {
+      results.aggregatedResults = Object.fromEntries(
+        Object.keys(results.results).map((name) => [
+          name,
+          { [results.results[name]]: 1 },
+        ])
+      );
+
+      return;
+    }
+
+    for (const name in results.children) constructTree(results.children[name]);
+
+    for (const name in results.children) {
+      const childResults = results.children[name];
+      results.aggregatedResults = Object.keys(
+        childResults.aggregatedResults
+      ).reduce((acc, mode) => {
+        if (!(mode in acc)) acc[mode] = {};
+        const modeAcc = acc[mode];
+        const stats = childResults.aggregatedResults[mode];
+        for (const name in stats) {
+          if (name in modeAcc) modeAcc[name] += stats[name];
+          else modeAcc[name] = stats[name];
+        }
+        return acc;
+      }, results.aggregatedResults || {});
+    }
+  }
+
+  // Now do another pass and aggregate the results.
+  let results = {
+    children: resultsObject,
+    aggregatedResults: {},
+  };
+  constructTree(results);
+  tree = results;
+
+  resultsNode = document.getElementById("results");
+  summaryLabel = document.getElementById("summary");
+  summaryStatusLabel = document.getElementById("summary-status");
+  leafTreeNodeTemplate = document.getElementById("leaf-tree-node-template");
+  nonLeafTreeNodeTemplate = document.getElementById(
+    "nonleaf-tree-node-template"
+  );
+
+  // Now make a nice lazy-loaded collapsible tree in `resultsNode`.
+  generateChildren(resultsNode);
+
+  summaryStatusLabel.classList.remove("hidden");
+
+  summaryLabel.onclick = () => {
```

Comment:
accessibility of this probably sucks but should be ok for a first revision

Reply:
I tried making a "back" button, but it looked so damn ugly I just scrapped it and decided to let someone else do it :P